server/backend/api/app/crud/crud_application.py

```diff
@@ -108,53 +106,76 @@ def get_application_roles(
 
 
 def get_application_role_assignments(
-    db: Session, application_id: int
+    db: Session, application_id: int, requester: schemas.Requester
 ) -> List[models.FamUserRoleXref]:
     """query the user / role cross reference table to retrieve the role
-    assignments
+    assignments.
+    Delegated Admin will only see user role assignments by the roles granted for them.
+    BCeID Delegated Admin will be further restricted for the same organization.
 
     :param db: database session
-    :type db: Session
-    :param application_id: the application id that we want to retrieve the role
-        assignments for.
-    :return: the role assignments for the given application
-
+    :param application_id: the application id to retrieve the role assignments for.
+    :param requester: the user who perform this request/action.
+    :return: the user role assignments for the given application.
     """
-    LOGGER.debug(f"Query for user role assignments for app id: {application_id}")
+    LOGGER.debug(f"Querying for user role assignments on app id:\
+                  {application_id} by requester: {requester} ")
 
-    crossref = (
+    # base query - users assigned to the application. This could be the case
+    #              for [APP]_ADMIN.
+    q = (
         db.query(models.FamUserRoleXref)
         .join(models.FamRole)
         .filter(models.FamRole.application_id == application_id)
-        .all()
     )
 
-    LOGGER.debug(f"Query for user role assignment complete with # of results = {len(crossref)}")
-    return crossref
-
-
-def get_application_id_by_role_id(db: Session, role_id):
-    # question: why we need to go to fam_application table?
-    # the fam_role table has the application_id already
-
-    application_id =\
-        db.query(models.FamApplication.application_id)\
-        .join(models.FamRole)\
-        .filter(models.FamRole.role_id == role_id)\
-        .scalar()
-
-    return application_id
-
+    if not crud_utils.is_app_admin(
+        application_id=application_id,
+        access_roles=requester.access_roles,
+        db=db
+    ):
+        # subquery for finding out what roles (role_ids) the requester
+        # (as an application delegated admin) is managing at.
```

Comment:
shall we add like ".....find out what roles the requester is managing at **for this specific application**"?

Reply:
Sure, that's more clear, added :)